pkg/prometheus/resource_selector.go

```diff
@@ -841,3 +846,29 @@ func (rs *ResourceSelector) validateEC2SDConfigs(ctx context.Context, sc *monito
 	}
 	return nil
 }
+
+func (rs *ResourceSelector) validateAzureSDConfigs(ctx context.Context, sc *monitoringv1alpha1.ScrapeConfig) error {
+	for i, config := range sc.Spec.AzureSDConfigs {
+		// Since Prometheus uses default authentication method as "OAuth"
+		if config.AuthenticationMethod == nil || *config.AuthenticationMethod == "OAuth" {
+			if config.TenantID == nil || len(*config.TenantID) == 0 {
+				return fmt.Errorf("[%d]: configuration requires a tenantID", i)
+			}
+
+			if config.ClientID == nil || len(*config.ClientID) == 0 {
+				return fmt.Errorf("[%d]: configuration requires a clientID", i)
+			}
+
+			if config.ClientSecret == nil {
+				return fmt.Errorf("[%d]: configuration requires a clientSecret", i)
+			}
+
+			if config.ClientSecret != nil {
+				if _, err := rs.store.GetSecretKey(ctx, sc.GetNamespace(), *config.ClientSecret); err != nil {
+					return fmt.Errorf("[%d]: %w", i, err)
+				}
+			}
```

Comment:
(nit) ClientSecret is checked for nil before.
```suggestion
			if _, err := rs.store.GetSecretKey(ctx, sc.GetNamespace(), *config.ClientSecret); err != nil {
				return fmt.Errorf("[%d]: %w", i, err)
			}
		}
```